d2l-hm-constants-behavior.js

```diff
@@ -365,3 +365,15 @@ if (typeof module !== 'undefined') {
 		Rels: D2LHypermediaRels
 	}
 }
```

Comment:
The modulizer interestingly took `d2l-hm-constants-behavior.html` and brought the contents of `index.js` into here. I think having everything in one `.js` file is preferred anyway... kind of neat that it did this.

Reply:
One thing that might be a problem with this approach is that we also use the `index.js` file in other places as just a vanilla script - e.g. [in the user-graph-service](https://github.com/Brightspace/user-graph-service/blob/master/src/graphql/modules/activities/UserResolver.js#L3). My intuition is that this won't play nicely with that use case anymore?

Reply:
Right, it won't. So... the correct solution then is likely to convert this "behavior" into just a normal JavaScript library that `export`s a bunch of things that projects can `import`, instead of relying on `D2L.Hypermedia.Whatever` globals.

I can do that now, which would turn this into a breaking change for consumers. Either way it'll break for someone, so probably worth it?

Reply:
Sounds reasonable to me! I suspect this is one of those cases where this maybe shouldn't have been a behaviour anyway, but hindsight is 20/20. As you said, the breaking change is going to happen anyway, so I'd say doing the most-correct breaking change seems like a good path 😄